Test failure: `TomcatServiceDestroyedConnectorTest.serviceUnavailableAfterConnectorIsDestroyed()`

```
TomcatServiceDestroyedConnectorTest > serviceUnavailableAfterConnectorIsDestroyed() FAILED
    java.lang.AssertionError: 
    Expecting:
     <200 OK>
    and actual:
     <503 Service Unavailable>
    to refer to the same object
        at com.linecorp.armeria.server.tomcat.TomcatServiceDestroyedConnectorTest.serviceUnavailableAfterConnectorIsDestroyed(TomcatServiceDestroyedConnectorTest.java:55)
```

--- a/tomcat9/src/main/java/com/linecorp/armeria/server/tomcat/TomcatService.java
+++ b/tomcat9/src/main/java/com/linecorp/armeria/server/tomcat/TomcatService.java
@@ -329,7 +329,7 @@ public abstract class TomcatService implements HttpService {
     @Override
     public final HttpResponse serve(ServiceRequestContext ctx, HttpRequest req) throws Exception {
         final Connector connector = connector();
-        if (connector == null || isConnectorStopped(connector.getState())) {
+        if (connector == null || !isConnectorAvailable(connector.getState())) {
             return HttpResponse.of(HttpStatus.SERVICE_UNAVAILABLE);
         }
         final Adapter coyoteAdapter = connector.getProtocolHandler().getAdapter();
@@ -347,7 +347,7 @@ public abstract class TomcatService implements HttpService {
                     }
                     return null;
                 }
-                if (isConnectorStopped(connector.getState())) {
+                if (!isConnectorAvailable(connector.getState())) {
                     if (res.tryWrite(ResponseHeaders.of(HttpStatus.SERVICE_UNAVAILABLE))) {
                         res.close();
                     }
@@ -375,7 +375,7 @@ public abstract class TomcatService implements HttpService {
                         return;
                     }
 
-                    if (isConnectorStopped(connector.getState())) {
+                    if (!isConnectorAvailable(connector.getState())) {
                         if (res.tryWrite(ResponseHeaders.of(HttpStatus.SERVICE_UNAVAILABLE))) {
                             res.close();
                         }
@@ -410,12 +410,11 @@ public abstract class TomcatService implements HttpService {
         return res;
     }
 
-    private static boolean isConnectorStopped(LifecycleState connectorState) {
+    private static boolean isConnectorAvailable(LifecycleState connectorState) {
         switch (connectorState) {
-            case STOPPED:
-            case DESTROYING:
-            case DESTROYED:
-            case FAILED:
+            case STARTED:
+            case STOPPING_PREP:
+            case STOPPING:
                 return true;
             default:
                 return false;

--- a/tomcat9/src/test/java/com/linecorp/armeria/server/tomcat/TomcatServiceDestroyedConnectorTest.java
+++ b/tomcat9/src/test/java/com/linecorp/armeria/server/tomcat/TomcatServiceDestroyedConnectorTest.java
@@ -16,6 +16,7 @@
 package com.linecorp.armeria.server.tomcat;
 
 import static org.assertj.core.api.Assertions.assertThat;
+import static org.awaitility.Awaitility.await;
 
 import java.io.File;
 
@@ -25,7 +26,6 @@ import org.junit.jupiter.api.Test;
 import org.junit.jupiter.api.extension.RegisterExtension;
 
 import com.linecorp.armeria.client.WebClient;
-import com.linecorp.armeria.common.AggregatedHttpResponse;
 import com.linecorp.armeria.common.HttpStatus;
 import com.linecorp.armeria.internal.testing.webapp.WebAppContainerTest;
 import com.linecorp.armeria.server.ServerBuilder;
@@ -51,11 +51,11 @@ class TomcatServiceDestroyedConnectorTest {
     @Test
     void serviceUnavailableAfterConnectorIsDestroyed() throws LifecycleException {
         final WebClient client = WebClient.of(server.httpUri());
-        AggregatedHttpResponse join = client.get("/api/").aggregate().join();
-        assertThat(join.status()).isSameAs(HttpStatus.OK);
+        assertThat(client.get("/api/").aggregate().join().status()).isSameAs(HttpStatus.OK);
         tomcatWithWebApp.stop();
         tomcatWithWebApp.getConnector().destroy();
-        join = client.get("/api/").aggregate().join();
-        assertThat(join.status()).isSameAs(HttpStatus.SERVICE_UNAVAILABLE);
+        // Use await because it can take time to destroy the connector.
+        await().untilAsserted(() -> assertThat(client.get("/api/").aggregate().join().status())
+                .isSameAs(HttpStatus.SERVICE_UNAVAILABLE));
     }
 }

--- a/tomcat9/src/test/java/com/linecorp/armeria/server/tomcat/TomcatServiceDestroyedConnectorTest.java
+++ b/tomcat9/src/test/java/com/linecorp/armeria/server/tomcat/TomcatServiceDestroyedConnectorTest.java
@@ -23,7 +23,6 @@ import java.io.File;
 import org.apache.catalina.LifecycleException;
 import org.apache.catalina.startup.Tomcat;
 import org.junit.jupiter.api.RepeatedTest;
-import org.junit.jupiter.api.Test;
 import org.junit.jupiter.api.extension.RegisterExtension;
 
 import com.linecorp.armeria.client.WebClient;

--- a/tomcat9/src/test/java/com/linecorp/armeria/server/tomcat/TomcatServiceDestroyedConnectorTest.java
+++ b/tomcat9/src/test/java/com/linecorp/armeria/server/tomcat/TomcatServiceDestroyedConnectorTest.java
@@ -16,13 +16,12 @@
 package com.linecorp.armeria.server.tomcat;
 
 import static org.assertj.core.api.Assertions.assertThat;
-import static org.awaitility.Awaitility.await;
 
 import java.io.File;
 
 import org.apache.catalina.LifecycleException;
 import org.apache.catalina.startup.Tomcat;
-import org.junit.jupiter.api.RepeatedTest;
+import org.junit.jupiter.api.Test;
 import org.junit.jupiter.api.extension.RegisterExtension;
 
 import com.linecorp.armeria.client.WebClient;
@@ -46,21 +45,14 @@ class TomcatServiceDestroyedConnectorTest {
             tomcatWithWebApp.start();
             sb.serviceUnder("/api/", TomcatService.of(tomcatWithWebApp.getConnector()));
         }
-
-        @Override
-        protected boolean runForEachTest() {
-            return true;
-        }
     };
 
-    @RepeatedTest(100)
+    @Test
     void serviceUnavailableAfterConnectorIsDestroyed() throws LifecycleException {
         final WebClient client = WebClient.of(server.httpUri());
         assertThat(client.get("/api/").aggregate().join().status()).isSameAs(HttpStatus.OK);
         tomcatWithWebApp.stop();
         tomcatWithWebApp.getConnector().destroy();
-        // Use await because it can take time to destroy the connector.
-        await().untilAsserted(() -> assertThat(client.get("/api/").aggregate().join().status())
-                .isSameAs(HttpStatus.SERVICE_UNAVAILABLE));
+        assertThat(client.get("/api/").aggregate().join().status()).isSameAs(HttpStatus.SERVICE_UNAVAILABLE);
     }
 }

--- a/tomcat9/src/test/java/com/linecorp/armeria/server/tomcat/TomcatServiceDestroyedConnectorTest.java
+++ b/tomcat9/src/test/java/com/linecorp/armeria/server/tomcat/TomcatServiceDestroyedConnectorTest.java
@@ -22,6 +22,7 @@ import java.io.File;
 
 import org.apache.catalina.LifecycleException;
 import org.apache.catalina.startup.Tomcat;
+import org.junit.jupiter.api.RepeatedTest;
 import org.junit.jupiter.api.Test;
 import org.junit.jupiter.api.extension.RegisterExtension;
 
@@ -46,9 +47,14 @@ class TomcatServiceDestroyedConnectorTest {
             tomcatWithWebApp.start();
             sb.serviceUnder("/api/", TomcatService.of(tomcatWithWebApp.getConnector()));
         }
+
+        @Override
+        protected boolean runForEachTest() {
+            return true;
+        }
     };
 
-    @Test
+    @RepeatedTest(100)
     void serviceUnavailableAfterConnectorIsDestroyed() throws LifecycleException {
         final WebClient client = WebClient.of(server.httpUri());
         assertThat(client.get("/api/").aggregate().join().status()).isSameAs(HttpStatus.OK);

